crates/wasi-provider/src/lib.rs

```diff
@@ -172,12 +172,11 @@ impl Provider for WasiProvider {
                 pod_name: pod,
                 container_name: container,
             })?;
-        Ok(handle
-            .0
-            .by_ref()
-            .lines()
-            .map(|l| l.unwrap_or_else(|_| "<error while reading line>".to_string()))
-            .collect::<Vec<String>>())
+        let mut output = Vec::default();
```

Comment:
`Vec::new` is shorter and the same thing as `Vec::default`. 

It's probably not worth it, but we could read the file metadata and find the length of the file and preallocate that much space in the Vec using `Vec::with_capacity`.